src/ui/pages/RequestAccounts/RequestAccounts.tsx

```diff
@@ -260,29 +272,52 @@ function RequestAccountsView({
   );
 }
 
+class NoWalletForEcosystemError extends Error {}
+
 export function RequestAccounts() {
   const [params] = useSearchParams();
   const origin = params.get('origin');
   const windowId = params.get('windowId');
+  const ecosystem = (params.get('ecosystem') || 'evm') as BlockchainType;
+  assertKnownEcosystems([ecosystem]);
 
   invariant(origin, 'origin get-parameter is required');
   invariant(windowId, 'windowId get-parameter is required');
 
-  const walletGroupsQuery = useQuery({
-    queryKey: ['wallet/uiGetWalletGroups'],
-    queryFn: () => walletPort.request('uiGetWalletGroups'),
-    useErrorBoundary: true,
-  });
-  const {
-    data: wallet,
-    isLoading,
-    isError,
-  } = useQuery({
-    queryKey: ['wallet/uiGetCurrentWallet'],
-    queryFn: () => {
-      return walletPort.request('uiGetCurrentWallet');
+  const { data, isLoading, isError, error } = useQuery({
+    retry: 1,
```

Comment:
why do we need this?

Reply:
This just reduces the default number of retries to `1`